src/Faker/Provider/pl_PL/Person.php

```diff
@@ -85,7 +85,7 @@ class Person extends \Faker\Provider\Person
      *
      * @var string
      */
-    protected static $title = array('mgr','inż.', 'dr', 'doc.');
+    protected static $title = array('mgr','inż.', 'dr', 'doc.', 'mgr. inż.', 'prof', 'dr. hab.');
```

Comment:
You should also fix the `$maleNameFormats` and `$femaleNameFormats` properties, because they might use a combination of two titles.